client/components/AllProducts.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'; 
+import {connect} from 'react-redux';
+import {fetchProducts} from '../store/products';
+import {Link} from 'react-router-dom';
+
+class AllProducts extends React.Component{
+    constructor(props){
+        super(props)
+    }
+    componentDidMount(){
+        this.props.fetchInitialData();
+    }
+    render(){
+        return (
+            <div>
+            <ul>
+             {this.props.products && this.props.products.map(product =>
+                <li key={product.id}> <Link to ={`/products/${product.id}`}>{product.id}</Link></li>
+            )}
+            </ul>
+            </div>
+        )
+    }
+}
+const mapStateToProps = state => (
+    {
+        products: state.products
+    })
+
+const mapDispatchToProps = dispatch => ({
+    fetchInitialData: () => {
```

Comment:
I would keep this on one line and get rid of the curly braces but up to you guys:
`fetchInitialData: () => dispatch(fetchProducts())`